File: AppointmentSchedulerWeb/src/view-model/business-entities/notification/notification.ts
import { NotificationType } from "../types/notification.types";
import { AppointmentNotification } from "./appointment-notification";
import { GeneralNotification } from "./general-notification";
import { SystemNotification } from "./system-notification";

export interface NotificationBase {
  createdAt: Date,
  uuid: string,
  message: string,
  type: NotificationType
}


export type AppNotification = NotificationBase | GeneralNotification | SystemNotification | AppointmentNotification;

File: AppointmentSchedulerAPI/layers/BusinessLogicLayer/Model/Notification.cs
using AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model.Types.Notification;

namespace AppointmentSchedulerAPI.layers.BusinessLogicLayer.Model
{
    public abstract class Notification
    {
        public DateTime? CreatedAt { get; set; } = DateTime.UtcNow;
        public int? Id { get; set; }
        public Guid? Uuid { get; set; }
        public NotificationType Type { get; set; }
        public required string Message { get; set; }
        public required NotificationOptions Options { get; set; }
        public required HashSet<NotificationRecipient> Recipients { get; set; }
    }
}